kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala

```diff
@@ -342,6 +343,31 @@ object ServiceDiscovery extends Logging {
     }
     serviceNode
   }
+
+  def withLock(
+      zkClient: CuratorFramework,
+      lockPath: String,
+      lockTimeOutSeconds: Long)(f: => Unit): Unit = {
+    var lock: InterProcessSemaphoreMutex = null
+    try {
```

Comment:
Can we simply use `scala.util.Using`? @pan3793

Reply:
yeap

Reply:
`scala.util.Using` was introduced in scala-2.13 ?

Reply:
we back ported it

Reply:
Can we use this instead ?
  `def using[A <: {def close(): Unit}](r: => A)(f: A => Unit): Unit = {`
    `val resource = r`
    `try { f(resource) } finally { if (resource != null) r.close() }`
  `}`

Reply:
`InterProcessSemaphoreMutex` is not a AutoCloseable, so  we need to add this in object `Releasable` ? 
`type AutoReleasableIsReleasable = {def release() : Unit}`
`implicit object AutoReleasableIsReleasable extends Releasable[{def release() : Unit}] {`
      `def release(resource: AutoReleasableIsReleasable): Unit = resource.release()`
`    }`
Or is there a better way? @pan3793

Reply:
Oops, I didn't realize that `InterProcessSemaphoreMutex` is not a `AutoCloseable`, the extension is feasible, but don't do it(at least don't modify in `Using.scala` file), keep the `Using` compatible with future Scala versions.

Reply:
Just leave it as it is.